cogs/dotastats.py

```diff
@@ -252,6 +254,47 @@ def __init__(self, bot):
 		self.chat_wheel_info = dotabase.get_chat_wheel_infos()
 		self.dota_gif_lock = asyncio.Lock()
 
+	async def get_meta_json(self): 
+		url = 'https://api.opendota.com/api/herostats'
+		async with httpx.AsyncClient() as client:
+			r = await client.get(url)
+		return r.json()
+
+	def sort_meta(self, json, num=10): 
+		"""re-orders the meta json based on pick/ban + winrate.
+		num = number of top heroes to include """
+		total_games = get_total_pro_games(json)
+		sorted_json = sorted(
+			json, 
+			reverse=True, 
+			# sorts by (winrate) + (pick/ban rate)
+			key=lambda x: (((x.get('pro_pick', 0) + x.get('pro_ban', 0)) / total_games)
+			+ ((x.get('pro_win', 0) / x.get('pro_pick', 1)))),
+			)
+		if num > 0: 
+			return sorted_json[:num]
+		return sorted_json
+
+	async def print_meta(self, ctx, num_to_list): 
+		"""prints the meta table"""
+		json = await self.get_meta_json()
+		sorted_json = self.sort_meta(json, num_to_list)
+		description = ("Top %s meta hero(s) in professional matches" % num_to_list)
```

Comment:
This works as a way to insert stuff into strings, but to keep it more consistant with the rest of the bot, I'd suggest changing this to use the string interpolation python provides which would be like this: `f"Top {num_to_list} meta hero(s) in professional matches"`

Reply:
updated everything, should be pushed